### ISSUE TITLE
Authetication process is triggered twice

### ISSUE THREAD

**Issue by anthonyraymond** [ahus1/keycloak-dropwizard-integration#6]
Hi, i would like to know if you are aware that the authentication process is triggered twice.

```
KeycloakJettyAuthenticator keycloak = new KeycloakDropwizardAuthenticator();
keycloak.setAdapterConfig(getKeycloakConfiguration(configuration));
ConstraintSecurityHandler securityHandler = new ConstraintSecurityHandler();
environment.getApplicationContext().setSecurityHandler(securityHandler);
environment.getApplicationContext().getSecurityHandler().setAuthenticator(keycloak);
```
trigger it once for every incomming request.

and in file `KeycloakAuthFilter` line 109:
`AuthOutcome outcome = authenticator.authenticate();`
trigger once again.

Is it a wanted behavior?

(The process can be followed in you enable TRACE loggin in the application)

**Comment by ahus1** [comment]
KeycloakDropwizardAuthenticator will trigger authentication in some, but not in all cases. As you point out authenticate() might be called again. I've added a check if authentication has already been performed. 

Please give 0.7.3-SNAPSHOT from https://oss.sonatype.org/content/repositories/snapshots a try. My existing tests are still "green" with this version. 

If you still see it being called twice, please provide a test case to describe your setup.

**Comment by anthonyraymond** [comment]
I just tryed it,and it works as expected.
thanks for the fix.

**Comment by ahus1** [comment]
This has been released as part of release 0.7.3 and should be available on maven central soon.

**Comment by anthonyraymond** [comment]
Glad to hear :wink: thanks for the fix
